contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/HttpRequestContext.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.http.core;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opentelemetry.distributedcontext.DistributedContext;
+import io.opentelemetry.trace.Span;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
+
+/** This class provides storage per request context on http client and server. */
+public class HttpRequestContext {
```

Comment:
should this be implemented via AutoValue?

Reply:
Don't know why this would be needed

Reply:
Yeah, just wanted to throw it out as a thought. Certainly not a requirement.